build/teamcity/cockroach/nightlies/microbenchmark_weekly.sh

```diff
@@ -120,6 +122,26 @@ done
   -- "$TEST_ARGS" \
   || exit_status=$?
 
+# Write metadata to a file for each set of benchmarks
+declare -A metadata
+metadata["run-time"]=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+metadata["baseline-commit"]=$baseline_commit
+metadata["benchmarks-commit"]=$benchmarks_commit
+metadata["machine"]=$GCE_MACHINE_TYPE
+metadata["goarch"]=amd64
+metadata["goos"]=linux
+metadata["repository"]=cockroach
+echo "" > "$output_dir/baseline/metadata.log"
+for key in "${!metadata[@]}"; do
+  echo "$key": "${metadata[$key]}" >> "$output_dir/baseline/metadata.log"
+done
+
+metadata["experiment-commit"]=${sha_arr[0]}
+metadata["experiment-commit-time"]=$(git show -s --format=%cI "${sha_arr[0]}")
+for key in "${!metadata[@]}"; do
+  echo "$key": "${metadata[$key]}" >> "$output_dir/experiment/metadata.log"
+done
```

Comment:
Is the idea that this metadata would be parsed by `roachprod-microbench` in the future? Might be worth having that tool generate the metadata in a well defined format (e.g., JSON) via a data structure in code, rather than writing it in the bash script.

Reply:
Yes, it's already part of this PR https://github.com/cockroachdb/cockroach/pull/129074. The format is that of the traditional go benchmark logs, and it's what [benchfmt](https://pkg.go.dev/golang.org/x/perf/storage/benchfmt) will consume to add `residues` (as they call it) to the benchmark results. Hence I didn't feel the need to have a tool generate it, but I can add a comment to mention here that these metadata files just become part of the benchmark logs that the reader will consume.